src/olympia/api/models.py

```diff
@@ -101,3 +106,38 @@ def generate_secret(byte_length):
                 '{} is too short; secrets must be longer than 32 bytes'
                 .format(byte_length))
         return force_text(binascii.b2a_hex(os.urandom(byte_length)))
+
+
+class APIKeyConfirmation(ModelBase):
+    user = models.OneToOneField(
+        UserProfile, primary_key=True, on_delete=models.CASCADE)
+    token = models.CharField(max_length=20)
+    confirmed_once = models.BooleanField(default=False)
+
+    @staticmethod
+    def generate_token():
+        """
+        Generate token for API Key Confirmation mechanism.
+
+        Returns a random 20 characters string (using a-z, A-Z, 0-9 characters).
+        """
+        return get_random_string(20)
+
+    def send_confirmation_email(self):
+        context = {
+            'api_key_confirmation_link': (
+                absolutify(reverse('devhub.api_key')) + f'?token={self.token}'
+            ),
+            'domain': settings.DOMAIN,
+        }
+        return send_mail_jinja(
+            'Confirmation for developer API keys',
```

Comment:
shouldn't that string be translated?

Reply:
It should yes, good catch.

Reply:
c5da7095f15a78ed9c11ac05ec65eeacd7a87eaf